logicrunner/logicrunner.go

```diff
@@ -1263,3 +1175,35 @@ func convertQueueToMessageQueue(queue []ExecutionQueueElement) []message.Executi
 
 	return mq
 }
+
+func (lr *LogicRunner) ClarifyPendingState(
+	ctx context.Context, es *ExecutionState, parcel core.Parcel,
+) error {
+	es.Lock()
+	if es.pending != message.PendingUnknown {
+		es.Unlock()
+		return nil
+	}
+
+	if parcel.Type() != core.TypeCallMethod {
+		es.Unlock()
+		es.pending = message.NotPending
+		return nil
+	}
+	es.Unlock()
+
+	has, err := lr.ArtifactManager.HasPendingRequests(ctx, es.Ref)
```

Comment:
few goroutines can go to AM, do we need it?

Reply:
what?

Reply:
we can use lock (not es.Lock) and only one goroutine will go over net to ledger, not all of them